akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1344,6 +1344,7 @@ private[stream] object Collect {
 
       private def pullIfNeeded(): Unit = {
         if (buffer.used < parallelism && !hasBeenPulled(in)) tryPull(in)
+        if (isClosed(in) && buffer.isEmpty) completeStage()
```

Comment:
Maybe for clarity since both cases are distinct

```suggestion
        if (isClosed(in) && buffer.isEmpty) completeStage()
        else if (buffer.used < parallelism && !hasBeenPulled(in)) tryPull(in)
        // else already pulled and waiting for next element
```